controller/volume_controller.go

```diff
@@ -2883,7 +2883,14 @@ func (c *VolumeController) checkAndFinishVolumeRestore(v *longhorn.Volume, e *lo
 		return err
 	}
 
-	if !isPurging && (v.Status.Robustness == longhorn.VolumeRobustnessHealthy || v.Status.Robustness == longhorn.VolumeRobustnessDegraded) && allScheduledReplicasIncluded {
+	degradedVolumeSupported, err := c.ds.GetSettingAsBool(types.SettingNameAllowVolumeCreationWithDegradedAvailability)
+	if err != nil {
+		return err
+	}
+
+	if !isPurging &&
+		(v.Status.Robustness == longhorn.VolumeRobustnessHealthy || v.Status.Robustness == longhorn.VolumeRobustnessDegraded) &&
+		(allScheduledReplicasIncluded || degradedVolumeSupported) {
```

Comment:
Sounds good to me. (nit: I feel `allScheduledReplicasIncluded` is redundant if we already know the volume is healthy)

However, suggest adjusting the condition to make it clear like below. WDYT? 

```go
	if !isPurging &&
		(v.Status.Robustness == longhorn.VolumeRobustnessHealthy && allScheduledReplicasIncluded) || (v.Status.Robustness == longhorn.VolumeRobustnessDegraded && degradedVolumeSupported) {
```

Also, we need to update DR doc to clarify the conditions for the activation.

Reply:
Modified. Thanks